apps/website/src/pages/courses/[courseSlug]/[unitNumber]/[[...chunkNumber]].tsx

```diff
@@ -2,20 +2,48 @@ import { useRouter } from 'next/router';
 import useAxios from 'axios-hooks';
 import { ErrorSection, ProgressDots, useAuthStore } from '@bluedot/ui';
 import { useEffect } from 'react';
-import UnitLayout from '../../../components/courses/UnitLayout';
-import { GetUnitResponse } from '../../api/courses/[courseSlug]/[unitNumber]';
-import { GetGroupDiscussionResponse } from '../../api/courses/[courseSlug]/[unitNumber]/groupDiscussion';
-import { GetCourseRegistrationResponse } from '../../api/course-registrations/[courseId]';
+import UnitLayout from '../../../../components/courses/UnitLayout';
+import { GetUnitResponse } from '../../../api/courses/[courseSlug]/[unitNumber]';
+import { GetGroupDiscussionResponse } from '../../../api/courses/[courseSlug]/[unitNumber]/groupDiscussion';
+import { GetCourseRegistrationResponse } from '../../../api/course-registrations/[courseId]';
 
-const CourseUnitPage = () => {
-  const { query: { courseSlug, unitNumber } } = useRouter();
+const CourseUnitChunkPage = () => {
+  const router = useRouter();
+  const {
+    query: {
+      courseSlug, unitNumber, chunkNumber, chunk,
+    },
+  } = router;
 
   if (typeof unitNumber !== 'string') {
     return <ProgressDots />;
   }
 
+  // Handle old ?chunk={n-1} format redirect
+  useEffect(() => {
+    if (typeof courseSlug === 'string' && typeof unitNumber === 'string' && typeof chunk === 'string') {
+      const oldChunkIndex = parseInt(chunk, 10);
+      if (!Number.isNaN(oldChunkIndex) && oldChunkIndex >= 0) {
+        const newChunkNumber = oldChunkIndex + 1;
+        router.replace(`/courses/${courseSlug}/${unitNumber}/${newChunkNumber}`);
+      }
+    }
+  }, [courseSlug, unitNumber, chunk, router]);
+
   const auth = useAuthStore((s) => s.auth);
 
+  let actualChunkNumber = '1';
+  // [[...chunkNumber]] catch-all syntax results in `chunkNumber` being an array, parse the first element
+  if (Array.isArray(chunkNumber) && chunkNumber.length > 0) {
+    const [firstChunk] = chunkNumber;
+    if (firstChunk) {
+      actualChunkNumber = firstChunk;
+    }
+  }
```

Comment:
If a link is missing the chunk `/1` at the end (e.g. `/courses/fish-test-course/1` rather than `/courses/fish-test-course/1/1`), it will use the first chunk, but it won't redirect the url.

I thought this would be better for SEO, so that links to the unit can all use `/courses/fish-test-course/1`, which doesn't appear to refer to the first chunk specifically.